client/session.go

```diff
@@ -735,7 +735,7 @@ func (s *session) Write(
 }
 
 func (s *session) writeAttempt(
-	namespace, id string,
+	nsID, tsID ts.ID,
```

Comment:
Yeah we need to be careful of the lifetime of these ts.IDs as well. 

We probably need to either copy the IDs using idPool to clone them (easiest) or make user supply a context, make them use the context to finalize the IDs (kind of annoying contract to have to know about), then keep our own context here, have their context depend on our context.